### PR TITLE
Create a window per screen

### PR DESCRIPTION
**Pull request by plfiorini**
A big window for all screens may hit hardware limits and it often has
a bad alignment as reported by #265.

Creating a window per screen doesn't suffer from these problems and,
as a nice side effects, it helps the Wayland port.

[ChangeLog][Greeter][Behavioral Change] Create a window per screen.

### CODE REVIEW

**Comment by plfiorini** [comment]
Resuming my work on this soon...

**Comment by plfiorini** [comment]
Updated... It no longer remove the screenModel let's see about that in a follow up patch.
The way it is done now it shouldn't change much for clone mode (which is what I get here).
I'm afraid clone mode might be default in the X world, so if this patch doesn't show any regression I'd like to merge it and take another look with Wayland.

**Comment by davidedmundson** [comment]
Did you have a chance to check this hasn't broken breeze before pushing?

File: src/greeter/GreeterApp.cpp [hunk 1]
```diff
@@ -134,35 +129,74 @@ namespace SDDM {
...
+        connect(view, &QQuickView::screenChanged, this, [view, this](QScreen *screen) {
+            if (screen == Q_NULLPTR)
+                removeViewForScreen(view);
+        });
+#endif
+
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 4, 0))
+        // always resize when the screen geometry changes
+        connect(screen, &QScreen::availableGeometryChanged, this, [view](const QRect &r) {
+            view->setGeometry(QRect(QPoint(0, 0), r.size()));
```
Comment (davidedmundson) [inline src/greeter/GreeterApp.cpp]: this isn't right. views geometry is global to the screen (in the X sense of the word) not to the the montitor.
If screen 2 size changes, you'll put it on top of screen 1.

view->setGeometry(screen->geometry()); will do you best

**Comment by plfiorini** [comment]
@davidedmundson yep breeze look fine here too and yes i won't remove the screen model for a while, in fact this PR reintroduces it.

File: src/greeter/GreeterApp.cpp (hunk 1 above)
Comment (plfiorini) [inline src/greeter/GreeterApp.cpp]: ops you are right, don't know why i left it this way

**Pull request merged**